File: retail industries Suite -all/fmcg_pro/fmcg_pro/fmcg_management/report/near_expiry_stock_alert/near_expiry_stock_alert.py
import frappe


def execute(filters=None):
    columns = [
        {"label": "Item", "fieldname": "item", "fieldtype": "Link",
         "options": "Item", "width": 150},
        {"label": "Batch", "fieldname": "batch", "fieldtype": "Link",
         "options": "Batch", "width": 120},
        {"label": "Warehouse", "fieldname": "warehouse", "fieldtype": "Link",
         "options": "Warehouse", "width": 140},
        {"label": "Expiry Date", "fieldname": "expiry_date", "fieldtype": "Date",
         "width": 110},
        {"label": "Days to Expiry", "fieldname": "days_to_expiry",
         "fieldtype": "Int", "width": 110},
        {"label": "Qty", "fieldname": "qty", "fieldtype": "Float", "width": 90},
    ]
    data = frappe.db.sql("""
        SELECT item, batch, warehouse, expiry_date, days_to_expiry, qty
        FROM `tabFMCG Batch Expiry Tracker`
        WHERE status IN ('Near Expiry', 'Expired')
        ORDER BY expiry_date ASC
    """, as_dict=1)
    return columns, data

File: retail industries Suite -all/fmcg_pro/fmcg_pro/fmcg_management/report/near_expiry_stock_alert/near_expiry_stock_alert.js
// Copyright (c) 2026, Your Organization and contributors
// For license information, please see license.txt

frappe.query_reports["Near Expiry Stock Alert"] = {
	filters: [
		{
			fieldname: "warehouse",
			label: __("Warehouse"),
			fieldtype: "Link",
			options: "Warehouse",
			default: "",
		},
		{
			fieldname: "item",
			label: __("Item"),
			fieldtype: "Link",
			options: "Item",
			default: "",
		},
		{
			fieldname: "status",
			label: __("Status"),
			fieldtype: "Select",
			options: "Near Expiry\nExpired",
			default: "Near Expiry",
		},
	],
};
